### PR TITLE
GitHub Action Updates & Dependabot Implementation

### PR DESCRIPTION
**Pull request by BillChirico**
- Update GitHub action versions
- Add dependabot package updates for `github-actions` and `nuget`
  - Added @dlamkins as auto assigner and reviewer for all of the automated PRs
  - Check for updates on a daily interval
  - **Note: Will need to create the labels assigned to the PRs, such as `dependencies`, `github-actions`, and `.NET` if you want them to be assigned to the PRs**

### CODE REVIEW

**Comment by dlamkins** [comment]
Running it to see if there are any deviations from our standard builds with the new action versions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,22 +12,22 @@ jobs:
     name: Nightly
     runs-on: windows-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3.0.2
         with:
           fetch-depth: 0
-      - uses: microsoft/setup-msbuild@v1.0.2
+      - uses: microsoft/setup-msbuild@v1.1.1
```
Comment (dlamkins) [inline .github/workflows/ci.yml]: > Error: Unable to resolve action `microsoft/setup-msbuild@v1.1.1`, unable to find version `v1.1.1`

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -7,22 +7,22 @@ jobs:
     name: Build
     runs-on: windows-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3.0.2
         with:
           fetch-depth: 0
-      - uses: microsoft/setup-msbuild@v1.0.2
+      - uses: microsoft/setup-msbuild@v1.1.1
```
Comment (dlamkins) [inline .github/workflows/pr.yml]: > Error: Unable to resolve action `microsoft/setup-msbuild@v1.1.1`, unable to find version `v1.1.1`

File: .github/workflows/tag.yml [hunk 3]
```diff
@@ -10,22 +10,22 @@ jobs:
     name: Nightly
     runs-on: windows-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3.0.2
         with:
           fetch-depth: 0
-      - uses: microsoft/setup-msbuild@v1.0.2
+      - uses: microsoft/setup-msbuild@v1.1.1
```
Comment (dlamkins) [inline .github/workflows/tag.yml]: > Error: Unable to resolve action `microsoft/setup-msbuild@v1.1.1`, unable to find version `v1.1.1`

**Review by dlamkins — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BillChirico) [reply .github/workflows/ci.yml]: Fixed version as per latest from https://github.com/marketplace/actions/setup-msbuild

File: .github/workflows/tag.yml (hunk 3 above)
Comment (BillChirico) [reply .github/workflows/tag.yml]: Fixed version as per latest from https://github.com/marketplace/actions/setup-msbuild

File: .github/workflows/pr.yml (hunk 2 above)
Comment (BillChirico) [reply .github/workflows/pr.yml]: Fixed version as per latest from https://github.com/marketplace/actions/setup-msbuild

**Review by dlamkins — APPROVED** [review]

**Pull request merged**